### PR TITLE
[rust] support handling settings command in Producer

### LINKED ISSUE #598: [Bug] Rust client Failed to parse config at client.handle_telemetry_command
### Before Creating the Bug Report

- [X] I found a bug, not just asking a question, which should be created in [GitHub Discussions](https://github.com/apache/rocketmq-clients/discussions).

- [X] I have searched the [GitHub Issues](https://github.com/apache/rocketmq-clients/issues) and [GitHub Discussions](https://github.com/apache/rocketmq-clients/discussions)  of this repository and believe that this is not a duplicate.

- [X] I have confirmed that this bug belongs to the current repository, not other repositories of RocketMQ.

### Programming Language of the Client

Rust

### Runtime Platform Environment

OS: Ubuntu 22.04

### RocketMQ Version of the Client/Server

Client Version: 5.0.0
Server Version: 5.1.3

### Run or Compiler Version

rustc (1.66.1+dfsg0ubuntu1-0ubuntu0.22.04.1)

### Describe the Bug

Sep 07 07:33:13.828 ERRO[/home/waa/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rocketmq-5.0.0/src/client.rs:196:33] handle telemetry command failed: Failed to parse config at client.handle_telemetry_command => receive telemetry command but there is no handler

Context:
    command: Settings(Settings { client_type: Some(Producer), access_point: None, backoff_policy: Some(RetryPolicy { max_attempts: 3, strategy: Some(ExponentialBackoff(ExponentialBackoff { initial: Some(Duration { seconds: 0, nanos: 10000000 }), max: Some(Duration { seconds: 1, nanos: 0 }), multiplier: 2.0 })) }), request_timeout: Some(Duration { seconds: 3, nanos: 0 }), user_agent: Some(Ua { language: Rust, version: "5.0.0", platform: "Ubuntu 22.04", hostname: "test1" }), metric: Some(Metric { on: false, endpoints: None }), pub_sub: Some(Publishing(Publishing { topics: [Resource { resource_namespace: "", name: "test123" }], max_body_size: 4194304, validate_message_type: true })) })
, component: client

### Steps to Reproduce

执行官方案例测试发送消息。  https://github.com/apache/rocketmq-clients/blob/master/rust/examples/producer.rs 

### What Did You Expect to See?

无

### What Did You See Instead?

无

### Additional Context

_No response_

### LINKED ISSUE #634: [Rust] failed to parse config at client.handle_telemetry_command sending messages
### Before Creating the Bug Report

- [X] I found a bug, not just asking a question, which should be created in [GitHub Discussions](https://github.com/apache/rocketmq-clients/discussions).

- [X] I have searched the [GitHub Issues](https://github.com/apache/rocketmq-clients/issues) and [GitHub Discussions](https://github.com/apache/rocketmq-clients/discussions)  of this repository and believe that this is not a duplicate.

- [X] I have confirmed that this bug belongs to the current repository, not other repositories of RocketMQ.

### Programming Language of the Client

Rust

### Runtime Platform Environment

win10 64

### RocketMQ Version of the Client/Server

rocketmq5.1.4

### Run or Compiler Version

rustc 1.75.0-nightly (e5cfc5547 2023-10-28)

[package]
name = "custrocketmq"
version = "0.1.0"
edition = "2021"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
rocketmq = "5.0.0"
tokio = { version = "1", features = ["full"] }

### Describe the Bug

1：broker 已经启动了proxy  
sh mqbroker -c /home/rocketmq/rocketmq-5.1.4/conf/broker.conf -pc /home/rocketmq/rocketmq-5.1.4/conf/rmq-proxy.json --enable-proxy

2：并使用官方例子examples/simple_producer.rs发送信息出现错误提示
ERRO[C:\Users\ljjav\.cargo\registry\src\rsproxy.cn-0dccff568467c15b\rocketmq-5.0.0\src\client.rs:196:33] handle telemetry command failed: Failed to parse config at client.handle_telemetry_command => receive telemetry command but there is no handler

详细错误提示：
thread 'main' panicked at C:\Users\ljjav\.cargo\registry\src\rsproxy.cn-0dccff568467c15b\rocketmq-5.0.0\src\client.rs:560:62:
called `Result::unwrap()` on an `Err` value: Failed to create session at session.create_session => failed to connect to peer

Context:
    peer: 172.24.0.4:8081

Source: transport error

Caused by:
    0: error trying to connect: deadline has elapsed
    1: deadline has elapsed

Stack backtrace:
   0: std::backtrace_rs::backtrace::dbghelp::trace
             at /rustc/e5cfc55477eceed1317a02189fdf77a4a98f2124/library\std\src\..\..\backtrace\src\backtrace\dbghelp.rs:98
   1: std::backtrace_rs::backtrace::trace_unsynchronized
             at /rustc/e5cfc55477eceed1317a02189fdf77a4a98f2124/library\std\src\..\..\backtrace\src\backtrace\mod.rs:66
   2: std::backtrace::Backtrace::create
             at /rustc/e5cfc55477eceed1317a02189fdf77a4a98f2124/library\std\src\backtrace.rs:331
   3: std::backtrace::Backtrace::capture
             at /rustc/e5cfc55477eceed1317a02189fdf77a4a98f2124/library\std\src\backtrace.rs:297
   4: anyhow::error::impl$1::from<tonic::transport::error::Error>
             at C:\Users\ljjav\.cargo\registry\src\rsproxy.cn-0dccff568467c15b\anyhow-1.0.75\src\error.rs:551
   5: core::convert::impl$3::into<tonic::transport::error::Error,anyhow::Error>
             at /rustc/e5cfc55477eceed1317a02189fdf77a4a98f2124\library\core\src\convert\mod.rs:757
   6: rocketmq::error::ClientError::set_source<tonic::transport::error::Error>
             at C:\Users\ljjav\.cargo\registry\src\rsproxy.cn-0dccff568467c15b\rocketmq-5.0.0\src\error.rs:122
   7: rocketmq::session::impl$1::new::async_fn$0::closure$0
             at C:\Users\ljjav\.cargo\registry\src\rsproxy.cn-0dccff568467c15b\rocketmq-5.0.0\src\session.rs:160
   8: enum2$<core::result::Result<tonic::transport::channel::Channel,tonic::transport::error::Error> >::map_err<tonic::transport::channel::Channel,tonic::transport::error::Error,rocketmq::error::ClientError,rocketmq::session::impl$1::new::async_fn$0::closure_en
             at /rustc/e5cfc55477eceed1317a02189fdf77a4a98f2124\library\core\src\result.rs:829
   9: rocketmq::session::impl$1::new::async_fn$0
             at C:\Users\ljjav\.cargo\registry\src\rsproxy.cn-0dccff568467c15b\rocketmq-5.0.0\src\session.rs:159
  10: rocketmq::session::impl$19::get_or_create_session::async_fn$0
             at C:\Users\ljjav\.cargo\registry\src\rsproxy.cn-0dccff568467c15b\rocketmq-5.0.0\src\session.rs:567
  11: rocketmq::client::impl$3::get_session_with_endpoints::async_fn$0
             at C:\Users\ljjav\.cargo\registry\src\rsproxy.cn-0dccff568467c15b\rocketmq-5.0.0\src\client.rs:342
  12: rocketmq::client::impl$3::receive_message::async_fn$0
             at C:\Users\ljjav\.cargo\registry\src\rsproxy.cn-0dccff568467c15b\rocketmq-5.0.0\src\client.rs:560
  13: rocketmq::simple_consumer::impl$0::receive_with::async_fn$0<ref$<str$> >
             at C:\Users\ljjav\.cargo\registry\src\rsproxy.cn-0dccff568467c15b\rocketmq-5.0.0\src\simple_consumer.rs:157
  14: rocketmq::simple_consumer::impl$0::receive::async_fn$0<alloc::string::String>
             at C:\Users\ljjav\.cargo\registry\src\rsproxy.cn-0dccff568467c15b\rocketmq-5.0.0\src\simple_consumer.rs:123
  15: custrocketmq::main::async_block$0
             at .\src\main.rs:68
  16: tokio::runtime::park::impl$4::block_on::closure$0<enum2$<custrocketmq::main::async_block_env$0> >
             at C:\Users\ljjav\.cargo\registry\src\rsproxy.cn-0dccff568467c15b\tokio-1.33.0\src\runtime\park.rs:282
  17: tokio::runtime::coop::with_budget
             at C:\Users\ljjav\.cargo\registry\src\rsproxy.cn-0dccff568467c15b\tokio-1.33.0\src\runtime\coop.rs:107
  18: tokio::runtime::coop::budget
             at C:\Users\ljjav\.cargo\registry\src\rsproxy.cn-0dccff568467c15b\tokio-1.33.0\src\runtime\coop.rs:73
  19: tokio::runtime::park::CachedParkThread::block_on<enum2$<custrocketmq::main::async_block_env$0> >
             at C:\Users\ljjav\.cargo\registry\src\rsproxy.cn-0dccff568467c15b\tokio-1.33.0\src\runtime\park.rs:282
  20: tokio::runtime::context::blocking::BlockingRegionGuard::block_on<enum2$<custrocketmq::main::async_block_env$0> >
             at C:\Users\ljjav\.cargo\registry\src\rsproxy.cn-0dccff568467c15b\tokio-1.33.0\src\runtime\context\blocking.rs:66
  21: tokio::runtime::scheduler::multi_thread::impl$0::block_on::closure$0<enum2$<custrocketmq::main::async_block_env$0> >
             at C:\Users\ljjav\.cargo\registry\src\rsproxy.cn-0dccff568467c15b\tokio-1.33.0\src\runtime\scheduler\multi_thread\mod.rs:87
  22: tokio::runtime::context::runtime::enter_runtime<tokio::runtime::scheduler::multi_thread::impl$0::block_on::closure_env$0<enum2$<custrocketmq::main::async_block_env$0> >,tuple$<> >    
             at C:\Users\ljjav\.cargo\registry\src\rsproxy.cn-0dccff568467c15b\tokio-1.33.0\src\runtime\context\runtime.rs:65
  23: tokio::runtime::scheduler::multi_thread::MultiThread::block_on<enum2$<custrocketmq::main::async_block_env$0> >
             at C:\Users\ljjav\.cargo\registry\src\rsproxy.cn-0dccff568467c15b\tokio-1.33.0\src\runtime\scheduler\multi_thread\mod.rs:86
  24: tokio::runtime::runtime::Runtime::block_on<enum2$<custrocketmq::main::async_block_env$0> >
             at C:\Users\ljjav\.cargo\registry\src\rsproxy.cn-0dccff568467c15b\tokio-1.33.0\src\runtime\runtime.rs:350
  25: custrocketmq::main
             at .\src\main.rs:76
  26: core::ops::function::FnOnce::call_once<void (*)(),tuple$<> >
             at /rustc/e5cfc55477eceed1317a02189fdf77a4a98f2124\library\core\src\ops\function.rs:250
  27: std::sys_common::backtrace::__rust_begin_short_backtrace<void (*)(),tuple$<> >
             at /rustc/e5cfc55477eceed1317a02189fdf77a4a98f2124\library\std\src\sys_common\backtrace.rs:154
  28: std::sys_common::backtrace::__rust_begin_short_backtrace<void (*)(),tuple$<> >
             at /rustc/e5cfc55477eceed1317a02189fdf77a4a98f2124\library\std\src\sys_common\backtrace.rs:154
  29: std::rt::lang_start::closure$0<tuple$<> >
             at /rustc/e5cfc55477eceed1317a02189fdf77a4a98f2124\library\std\src\rt.rs:167
  30: std::rt::lang_start_internal::closure$2
             at /rustc/e5cfc55477eceed1317a02189fdf77a4a98f2124/library\std\src\rt.rs:148
  31: std::panicking::try::do_call
             at /rustc/e5cfc55477eceed1317a02189fdf77a4a98f2124/library\std\src\panicking.rs:504
  32: std::panicking::try
             at /rustc/e5cfc55477eceed1317a02189fdf77a4a98f2124/library\std\src\panicking.rs:468
  33: std::panic::catch_unwind
             at /rustc/e5cfc55477eceed1317a02189fdf77a4a98f2124/library\std\src\panic.rs:142
  34: std::rt::lang_start_internal
             at /rustc/e5cfc55477eceed1317a02189fdf77a4a98f2124/library\std\src\rt.rs:148
  35: std::rt::lang_start<tuple$<> >
             at /rustc/e5cfc55477eceed1317a02189fdf77a4a98f2124\library\std\src\rt.rs:166
  36: main
  37: invoke_main
             at D:\a\_work\1\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:78
  38: __scrt_common_main_seh
             at D:\a\_work\1\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:288
  39: BaseThreadInitThunk
  40: RtlUserThreadStart

stack backtrace:
   0: std::panicking::begin_panic_handler
             at /rustc/e5cfc55477eceed1317a02189fdf77a4a98f2124/library\std\src\panicking.rs:597
   1: core::panicking::panic_fmt
             at /rustc/e5cfc55477eceed1317a02189fdf77a4a98f2124/library\core\src\panicking.rs:72
   2: core::result::unwrap_failed
             at /rustc/e5cfc55477eceed1317a02189fdf77a4a98f2124/library\core\src\result.rs:1653
   3: enum2$<core::result::Result<rocketmq::session::Session,rocketmq::error::ClientError> >::unwrap<rocketmq::session::Session,rocketmq::error::ClientError>
             at /rustc/e5cfc55477eceed1317a02189fdf77a4a98f2124\library\core\src\result.rs:1077
   4: rocketmq::client::impl$3::receive_message::async_fn$0
             at C:\Users\ljjav\.cargo\registry\src\rsproxy.cn-0dccff568467c15b\rocketmq-5.0.0\src\client.rs:560
   5: rocketmq::simple_consumer::impl$0::receive_with::async_fn$0<ref$<str$> >
             at C:\Users\ljjav\.cargo\registry\src\rsproxy.cn-0dccff568467c15b\rocketmq-5.0.0\src\simple_consumer.rs:157
   6: rocketmq::simple_consumer::impl$0::receive::async_fn$0<alloc::string::String>
             at C:\Users\ljjav\.cargo\registry\src\rsproxy.cn-0dccff568467c15b\rocketmq-5.0.0\src\simple_consumer.rs:123
   7: custrocketmq::main::async_block$0
             at .\src\main.rs:68
   8: tokio::runtime::park::impl$4::block_on::closure$0<enum2$<custrocketmq::main::async_block_env$0> >
             at C:\Users\ljjav\.cargo\registry\src\rsproxy.cn-0dccff568467c15b\tokio-1.33.0\src\runtime\park.rs:282
   9: tokio::runtime::coop::with_budget
             at C:\Users\ljjav\.cargo\registry\src\rsproxy.cn-0dccff568467c15b\tokio-1.33.0\src\runtime\coop.rs:107
  10: tokio::runtime::coop::budget
             at C:\Users\ljjav\.cargo\registry\src\rsproxy.cn-0dccff568467c15b\tokio-1.33.0\src\runtime\coop.rs:73
  11: tokio::runtime::park::CachedParkThread::block_on<enum2$<custrocketmq::main::async_block_env$0> >
             at C:\Users\ljjav\.cargo\registry\src\rsproxy.cn-0dccff568467c15b\tokio-1.33.0\src\runtime\park.rs:282
  12: tokio::runtime::context::blocking::BlockingRegionGuard::block_on<enum2$<custrocketmq::main::async_block_env$0> >
             at C:\Users\ljjav\.cargo\registry\src\rsproxy.cn-0dccff568467c15b\tokio-1.33.0\src\runtime\context\blocking.rs:66
  13: tokio::runtime::scheduler::multi_thread::impl$0::block_on::closure$0<enum2$<custrocketmq::main::async_block_env$0> >
             at C:\Users\ljjav\.cargo\registry\src\rsproxy.cn-0dccff568467c15b\tokio-1.33.0\src\runtime\scheduler\multi_thread\mod.rs:87
  14: tokio::runtime::context::runtime::enter_runtime<tokio::runtime::scheduler::multi_thread::impl$0::block_on::closure_env$0<enum2$<custrocketmq::main::async_block_env$0> >,tuple$<> >    
             at C:\Users\ljjav\.cargo\registry\src\rsproxy.cn-0dccff568467c15b\tokio-1.33.0\src\runtime\context\runtime.rs:65
  15: tokio::runtime::scheduler::multi_thread::MultiThread::block_on<enum2$<custrocketmq::main::async_block_env$0> >
             at C:\Users\ljjav\.cargo\registry\src\rsproxy.cn-0dccff568467c15b\tokio-1.33.0\src\runtime\scheduler\multi_thread\mod.rs:86
  16: tokio::runtime::runtime::Runtime::block_on<enum2$<custrocketmq::main::async_block_env$0> >
             at C:\Users\ljjav\.cargo\registry\src\rsproxy.cn-0dccff568467c15b\tokio-1.33.0\src\runtime\runtime.rs:350
  17: custrocketmq::main
             at .\src\main.rs:76
  18: core::ops::function::FnOnce::call_once<void (*)(),tuple$<> >
             at /rustc/e5cfc55477eceed1317a02189fdf77a4a98f2124\library\core\src\ops\function.rs:250
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
Oct 31 14:59:52.848 DEBG[C:\Users\ljjav\.cargo\registry\src\rsproxy.cn-0dccff568467c15b\rocketmq-5.0.0\src\client.rs:201:25] receive shutdown signal, stop heartbeat task and telemetry command handler, component: client
Oct 31 14:59:52.856 INFO[C:\Users\ljjav\.cargo\registry\src\rsproxy.cn-0dccff568467c15b\rocketmq-5.0.0\src\client.rs:206:13] heartbeat task and telemetry command handler are stopped, component: client
error: process didn't exit successfully: `target\debug\custrocketmq.exe` (exit code: 101)

 *  The terminal process "C:\Users\ljjav\.cargo\bin\cargo.exe 'run', '--package', 'custrocketmq', '--bin', 'custrocketmq'" terminated with exit code: 101. 
 *  Terminal will be reused by tasks, press any key to close it. 

### Steps to Reproduce

每次执行均出现此问题

### What Did You Expect to See?

如何解决此问题正常发送信息

### What Did You See Instead?

一直有问题

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by glcrazier**
### Which Issue(s) This PR Fixes

Fixes #634 
Fixes #598

### Brief Description

I've refactored the way of processing telemetry commands. Now the transaction command from the server is processed in Producer instead of Client.  The client with different types can handle telemetry command within its own impl block.

### How Did You Test This Change?

run cargo test and example to make sure it works

### CODE REVIEW

File: rust/src/producer.rs [hunk 1]
```diff
@@ -118,7 +196,53 @@ impl Producer {
...
+        command: pb::RecoverOrphanedTransactionCommand,
+        transaction_checker: &Option<Box<TransactionChecker>>,
+        endpoints: Endpoints,
+    ) -> Result<(), ClientError> {
+        let transaction_id = command.transaction_id;
+        let message = command.message.unwrap();
+        let message_id = message
+            .system_properties
+            .as_ref()
+            .unwrap()
```
[the remark this answers is not in the document]
Comment (glcrazier) [reply rust/src/producer.rs:210]: Got it, will fix

File: rust/src/producer.rs [hunk 2]
```diff
@@ -93,23 +116,80 @@ impl Producer {
...
+        let topics = option_guard.topics();
+        if let Some(topics) = topics {
             for topic in topics {
                 self.client.topic_route(topic, true).await?;
             }
         }
+        drop(option_guard);
+        let transaction_checker = self.transaction_checker.take();
+        if transaction_checker.is_some() {
+            self.transaction_checker = Some(Box::new(|_, _| TransactionResolution::UNKNOWN));
```
Comment (lizhanhui) [inline rust/src/producer.rs:152]: Why not simply take the transaction checker and leave None afterward?

Comment (glcrazier) [reply rust/src/producer.rs:152]: > Why not simply take the transaction checker and leave None afterward?

In `send_transaction_message` function, the producer uses `has_transaction_checker` function to check if it's legal to call this function. If we leave None, we have no idea whether it's a transaction producer.

Comment (lizhanhui) [reply rust/src/producer.rs:152]: You are right.

**Review by lizhanhui — APPROVED** [review]
Now things look good

**Pull request merged**